core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -27,12 +28,14 @@ import scala.language.postfixOps
 object MultiCloudWorkspaceService {
   def constructor(dataSource: SlickDataSource,
                   workspaceManagerDAO: WorkspaceManagerDAO,
+                  samDAO: SamDAO,
```

Comment:
Threading in a SamDAO here so we can make the check against the billing project for `createWorkspace` permissions.